webapp/WEB-INF/conf/core_context.xml

```diff
@@ -50,7 +53,10 @@
     <bean id="siteMessageHandler" class="fr.paris.lutece.portal.service.message.SiteMessageHandler"/>
 
     <!--                      Search Engine                                   -->
-    <bean id="searchEngine" class="fr.paris.lutece.portal.service.search.LuceneSearchEngine" />
+    <bean id="searchEngine" class="fr.paris.lutece.portal.service.search.LuceneSearchEngine">
+        <aop:scoped-proxy proxy-target-class="false" />
```

Comment:
Without it, the bean is injected directly, but we need a serializable proxy because XPage applications are stored in session. 

If you don't need session serialization, you can omit it. 


> Le 23 juil. 2015 à 12:41, Jon Harper <notifications@github.com> a écrit :
> 
> In webapp/WEB-INF/conf/core_context.xml:
> 
> > @@ -50,7 +53,10 @@
> >      <bean id="siteMessageHandler" class="fr.paris.lutece.portal.service.message.SiteMessageHandler"/>
> >  
> >      <!--                      Search Engine                                   -->
> > -    <bean id="searchEngine" class="fr.paris.lutece.portal.service.search.LuceneSearchEngine" />
> > +    <bean id="searchEngine" class="fr.paris.lutece.portal.service.search.LuceneSearchEngine">
> > +        <aop:scoped-proxy proxy-target-class="false" />
> According to http://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-factory-scopes-other-injection
> 
> You do not need to use the aop:scoped-proxy/ in conjunction with beans that are scoped as singletons or prototypes. It is an error to try to create a scoped proxy for a singleton bean (and the resulting BeanCreationException will certainly set you straight in this regard).
> 
> It's the same for the version we use (3.1.x). Did you add it to fix a problem ?
> 
> —
> Reply to this email directly or view it on GitHub.
>